fileMonitor/monitor.go

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright 2023 Nestybox, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package fileMonitor
+
+import (
+	"os"
+	"time"
+)
+
+type cmd int
+
+const (
+	stop cmd = iota
+)
+
+// Monitors files associated with the given FileMon instance
+func fileMon(fm *FileMon) {
```

Comment:
You may want to pass a context or a similar construct to stop this goroutine cleanly upon sysbox-mgr going down.